### PR TITLE
Added JUCEHelpers

### LINKED ISSUE #16: Implement toSIMDRegister
Includes creating a generic SIMD alias wrapper

### PR DESCRIPTION
**Pull request by JanosGit**
Instead of re-implementing the toSIMDRegister member function I went with moving juce specific functionality to a own set of free functions and implemented a few more along with that

### CODE REVIEW

File: include/vctr/Miscellaneous/JUCEHelpers.h [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Lesser General Public License version 3 for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    version 3 along with VCTR.  If not, see <https://www.gnu.org/licenses/>.
+  ==============================================================================
+*/
+
+namespace vctr::juce_helpers
```
Comment (psciri) [inline include/vctr/Miscellaneous/JUCEHelpers.h:23]: ah, I see the clang-tidy file already had its impact.. :D

File: include/vctr/Miscellaneous/JUCEHelpers.h [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    from a plugin host you cannot expect any guaranteed alignment. In this case, you should
+    use this function.
+
+    @see blockChannelAsSpanSIMDAligned
+ */
+template <class SampleType>
+auto blockChannelAsSpan (const juce::dsp::AudioBlock<SampleType>& block, size_t channelIdx)
+{
+    auto numSamples = block.getNumSamples();
+    return Span { block.getChannelPointer (channelIdx), numSamples };
```
Comment (psciri) [inline include/vctr/Miscellaneous/JUCEHelpers.h:93]: should this maybe assert that the channel index is within the number of available channels?

**Review by psciri — COMMENTED** [review]
Looking good to me! 👍

File: include/vctr/Miscellaneous/JUCEHelpers.h (hunk 2 above)
Comment (JanosGit) [reply include/vctr/Miscellaneous/JUCEHelpers.h:93]: `getChannelPointer` already should assert that internally so I don't see a point in catching that twice

**Review by psciri — APPROVED** [review]

File: include/vctr/Miscellaneous/JUCEHelpers.h [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+/** Returns a Span that views a single channel of a juce::dsp::AudioBlock.
+
+    This allows you to operate on audio channels with all available VCTR features.
+    In case the SampleType of the block is const, the Span will give you read access,
+    otherwise it will also allow you write access to the channel data.
+
+    Note: If you are working with audio blocks that view memory allocated by JUCE,
+    the allocated memory should be SIMD aligned. In that case consider using
+    blockChannelAsSpanSIMDAligned, which will return a Span that expects the memory
+    being SIMD aligned and might chose some better optimised versions of VCTR expressions
```
Comment (TheWaveWarden) [inline include/vctr/Miscellaneous/JUCEHelpers.h]: *choose 😋

**Review by TheWaveWarden — APPROVED** [review]
Looks good 👍

File: include/vctr/Miscellaneous/JUCEHelpers.h (hunk 3 above)
Comment (JanosGit) [reply include/vctr/Miscellaneous/JUCEHelpers.h]: Thx ;) Fixed!

**Pull request merged**